Public/Src/Engine/Processes/SandboxedProcessOutputBuilder.cs

```diff
@@ -102,11 +107,16 @@ public bool AppendLine(string data)
                 {
                     HandleRecoverableIOException(() => m_textWriter.WriteLine(data));
                 }
+                else if (m_fileStorage == null && m_stringBuilder.Length >= m_maxMemoryLength)
+                {
+                    // The caller should have configured an observer, called above. If not and no backing file configured,
+                    // we stop collecting the output stream at the in-memory buffer length, dropping any remainder.

```

Comment:
I'm maybe missing the context, but this silent drop looks a little dangerous. Shouldn't we return false here, or some other indication that the data was not written?

Reply:
Returning false breaks the stream in DetouredProcess. We actually just want to silently drop the data in this case, but note there is a call to the observer above. If there is no file redirection and no observer then the stream effectively goes to devnull

Reply:
Added an assert that there should be an observer or and output file or both to help avoid issues.